BasicGUI/QMLSources/main.qml

```diff
@@ -43,24 +45,59 @@ ApplicationWindow
         Menu 
         {
             title: qsTr("&View")
-            Action 
+            delegate: CheckBox{}
+
+            Action
             {
-                text: qsTr("&Show Image before trasformation...")
-                onTriggered:
+                checked: previev_is_active
+                checkable: true
+                text: "Preview selected image..."
+                onCheckedChanged: 
+                {
+                    if(previev_is_active == false)
+                    {   
+                        previev_is_active = true;
+                        if(image_selected)
+                        {
+                            loaded_image.visible = true;
+                        }
+                    }
+                    else 
+                    {
+                        previev_is_active = false;
+                        if(image_selected)
+                        {
+                            loaded_image.visible = false;
+                        }
+                    }
+                }
+                Component.onCompleted: 
                 {
-                       if(image_selected)
-                       {
-                        loaded_image.visible = true;
-                       }
+                    checked = preview_is_active
                 }
             }
-             MenuSeparator { }
-            Action 
+
+            Action
             {
-                text: qsTr("&HideImage before trasformation...")
-                onTriggered:
+                text: "Preview rendered image..."
+                checked: rendered_previev_is_active
```

Comment:
You need binding `rendered_preview_is_active: actionPreviewShow.checked`

Reply:
Corrected.

Reply:
Truly you changed code. Unfortunately you still manually flip boolean state.

Reply:
Corrected.